client/server/render/index.js

```diff
@@ -242,7 +242,7 @@ export function serverRender( req, res ) {
 
 		// And cache on the server, too.
 		if ( cacheKey ) {
-			const cacheableServerState = pick( context.store.getState(), cacheableReduxSubtrees );
+			const cacheableServerState = pick( context.store.getState(), reduxSubtrees );
```

Comment:
`cacheableReduxSubtrees` only contains `documentHead` and `themes` (and `ui`) subtrees never get included.
`reduxSubtrees` should also contain `themes` (and `ui`) for isomorphic sections.

Reply:
I don't know exactly why we only used a subsection of the state, but I'd guess it was done on purpose. A guess is that it uses less memory to only cache the parts of the tree we need?

What if we instead added the other stores we need to `cacheableReduxSubtrees`?

Hopefully others have good feedback too, as I don't know a lot about this!

Reply:
In any case, I'd do what @noahtallen suggests - continuing to use `cacheableReduxSubtrees` and if necessary, adding trees to that array. 

#13789 seems to fix a problem that was occurring when we start caching the `themes` and `ui` - it appears that caching server state for logged-in users was problematic. Have you checked that this PR doesn't reintroduce the same problem?

Reply:
@noahtallen @tyxla Thank you for your feedback!
> Have you checked that this PR doesn't reintroduce the same problem?

Indeed, my changes we're caching logged-in users accessing /theme pages.

I also agree, pushing to `cacheableReduxSubtrees` makes more sense.
I added the subtrees,  to a separate `cacheableLoggedOutReduxSubtrees` so things are clearer.

On the other hand, maybe it'd be better to have these subtrees configurable from the `client/sections.js` file (as a follow-up?). 
It took a bit of digging around to stumble on these.

Reply:
> On the other hand, maybe it'd be better to have these subtrees configurable from the client/sections.js file (as a follow-up?).

That sounds like a useful configuration option - is this something you're interested to try?

Reply:
> That sounds like a useful configuration option - is this something you're interested to try?
I'd be happy to do it.